src/clang.rs

```diff
@@ -187,7 +187,7 @@ impl Cursor {
 
     /// Is the referent a template specialization?
     pub fn is_template(&self) -> bool {
-        self.specialized().is_valid()
+        self.specialized().map_or(false, |c| c.is_valid())
```

Comment:
Checking `is_some()` will only work after my requested changes to the `is_valid()` checking. This is a good insight, though.